mailer.js

```diff
@@ -154,8 +154,12 @@ module.exports = function (log) {
     var location = message.location
     // construct the location string from the location object
     if (location) {
-      if (location.city) {
+      if (location.city && location.stateCode) {
+        return translator.format(translator.gettext('%(city)s, %(stateCode)s, %(country)s (estimated)'), location)
```

Comment:
Ack looks like you found a gap in our coverage. Mind adding test cases in for 

* city and country
* state and country
* only country